### PR TITLE
Fixes reuse compliance

### LINKED ISSUE #815: Fix reuse compliance
**How to categorize this issue?**

/area compliance
/kind enhancement
/kind cleanup

**What would you like to be added**:
After removal of the vendor folder dependencies were not cleaned-up from `.reuse/dep5`. This lead to the following non-compliance linter report attached with this issue.
There was a recommendation made by @RaphaelVogel in https://github.com/gardener/etcd-druid/pull/770 to remove the vendor directory and then adapt `.reuse/dep5` accordingly but it seems that we missed adapting it.

[reuse.log](https://github.com/user-attachments/files/15964719/reuse.log)

### PR DESCRIPTION
**Pull request by unmarshall**
**How to categorize this PR?**
/area compliance
/kind cleanup

**What this PR does / why we need it**:

1. Removes the vendor directory licenses from .reuse/dep5
2. Removes unused licenses

**Which issue(s) this PR fixes**:
Fixes #815 

**Special notes for your reviewer**:

**Release note**:

```other operator
NONE
```

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -2,7 +2,7 @@
 
 <image src="logo/etcd-druid-logo.png" style="width:300px"></image>
 
-[![REUSE status](https://api.reuse.software/badge/github.com/gardener/etcd-druid)](https://api.reuse.software/info/github.com/gardener/etcd-druid)
+[![REUSE status](https://api.reuse.software/badge/github.com/unmarshall/etcd-druid)](https://api.reuse.software/info/github.com/unmarshall/etcd-druid)
```
Comment (renormalize) [inline README.md:5]: This could be reverted to `gardener`.

**Review by renormalize — APPROVED** [review]

File: README.md (hunk 1 above)
Comment (unmarshall) [reply README.md]: done

**Review by RaphaelVogel — COMMENTED** [review]
/lgtm

**Review by RaphaelVogel — APPROVED** [review]

**Pull request merged**